README.md

```diff
@@ -77,17 +90,62 @@ Therefore I created this script to allow you to easily scan for and purge these
 
-## Usage Notes -READ THIS
+## **USAGE NOTES**
+
+1. **IF IT FREEZES** while scanning, it is probably because you clicked within the command prompt window and entered "selection mode" which pauses everything. **To unfreeze it, simply right click within the window, or press the Escape key.**
+
+2. I'm a total amateur, so if something doesn't work I'll try to fix it but might not even know how, so don't expect too much. Therefore **I OFFER NO WARRANTY OR GUARANTEE FOR THIS SCRIPT. USE AT YOUR OWN RISK.** I tested it on my own and implemented some failsafes as best as I could, but there could always be some kind of unexpected bug. You should inspect the code yourself.
+
+## Installation
+
+For windows, the best way to install is to download the latest EXE file from the [Releases page][Releases]. 
+
+For Mac/Linux the easiest way is to install from the install script
+
+```bash
+bash <(curl -s https://raw.githubusercontent.com/ThioJoe/YT-Spammer-Purge/main/install.sh)
+```
+
+Either way, you DO need to acquire your own API credentials file to access the YouTube API - [See Instructions Here][API Key].
+
+See one of the following if you need further help:
+* [Linux Setup Instructions][Instructions Linux]
+* [MacOS Setup Instructions][Instructions MacOS]
+
+### **Docker Instructions:**
+
+Before running `docker-compose` you must run the `YTSpammerPurge.py` script at least once with your `client_secrets.json` file to confirm OAuth credentials and generate the config/token files.
+
+The generated config files, token, and Spam Purge Resources will all be bound to the docker container via volumes.
+
+Once you generated the token and config files you are ready to run the docker image.
+
+
+Now you can run `docker-compose up` to start the container, or use the image to run on a Kubernetes cluster for example.
+To build your own version you can run this command: `docker-compose -f docker-compose.yml -f docker-compose.override.yml up --build`
+
+### Manual Installation
+
+See one of the following for help:
+* [Linux Setup Instructions][Instructions Linux]
+* [MacOS Setup Instructions][Instructions MacOS]
 
-1. To use this script, you will need to obtain your own API credentials file by making a project via the Google Developers Console (aka 'Google Cloud Platform'). The credential file should be re-named `client_secret.json` and be placed in the same directory as this script. [See Instructions Here][API Key].
+#### Updating
 
-2. **IF IT FREEZES** while scanning, it is probably because you clicked within the command prompt window and entered "selection mode" which pauses everything. **To unfreeze it, simply right click within the window, or press the Escape key.**
+If you have the exe installed you will need to redownload the entire exe.
+
+If you installed from source, run the `install.sh` file inside your installation to update
+
+
+## Instructions - Obtaining YouTube API Key
```

Comment:
My only suggestion would be to move the obtaining YT key higher, above update. It's an important step and one everyone will have to do to use this tool. Having it below update might hide it from people that don't read everything